### PR TITLE
Add initial service locator structure and utilities

### PR DESCRIPTION
**Pull request by devfrostybeard**
Added IService interface, ServiceLocator and TypeUtilities classes under new FrozenCircle plugin. These provide a boilerplate for ServiceLocator pattern that allows different parts of an application to be loosely coupled yet able to locate and use each other. IService interface ensures that all services will be Disposible while ServiceLocator provides mechanisms to register, unregister and get services. TypeUtilities will aid in creating instances and identifying types that extend certain interfaces.

### CODE REVIEW

File: Topdown RPG/Assets/Plugins/FrozenCircle/Core/Utils/IInitializable.cs [hunk 1]
```diff
@@ -0,0 +1,11 @@
+﻿
+using System.Threading.Tasks;
+
+namespace FrozenCircle.Core.Utils
+{
+    public interface IInitializable
+    {
+        // TODO: Refactor this to be UniTask
```
Comment (devfrostybeard) [inline Topdown RPG/Assets/Plugins/FrozenCircle/Core/Utils/IInitializable.cs:8]: Why is this TODO Here?

File: Topdown RPG/Assets/Plugins/FrozenCircle/Core/Services/ServiceLocator.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+using FrozenCircle.Core.Utils;
+using UnityEngine;
+
+namespace FrozenCircle.Core.Services
+{
+    public static class ServiceLocator
+    {
+        private static Dictionary<Type, IService> services = new Dictionary<Type, IService>();
+
+        [RuntimeInitializeOnLoadMethod]
```
Comment (kyle-ewing) [inline Topdown RPG/Assets/Plugins/FrozenCircle/Core/Services/ServiceLocator.cs:13]: Maybe use a different method?

File: Topdown RPG/Assets/Plugins/FrozenCircle/Core/Utils/TypeUtilities.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+namespace FrozenCircle.Core.Utils
+{
+    public static class TypeUtilities
+    {
+        public static List<Type> GetConcreteTypesThatExtend<T>()
```
Comment (devfrostybeard) [inline Topdown RPG/Assets/Plugins/FrozenCircle/Core/Utils/TypeUtilities.cs:10]: Documentation must be added

File: Topdown RPG/Assets/Plugins/FrozenCircle/Core/Utils/TypeUtilities.cs [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+                AppDomain
+                    .CurrentDomain
+                    .GetAssemblies()
+                    .SelectMany(x => x.GetTypes())
+                    .Where(x => !x.IsInterface && !x.IsAbstract && baseType.IsAssignableFrom(x))
+                    .ToList();
+            return assemblies;
+        } 
+        
+        public static List<Type> GetConcreteTypesThatExtend(Type baseType)
```
Comment (devfrostybeard) [inline Topdown RPG/Assets/Plugins/FrozenCircle/Core/Utils/TypeUtilities.cs:23]: Documentation must be added

**Review by devfrostybeard — COMMENTED** [review]
Ensure all documentation is fnished

**Review by kyle-ewing — CHANGES_REQUESTED** [review]

**Review by kyle-ewing — APPROVED** [review]

**Pull request merged**